Example/NYTViewController.m

```diff
@@ -23,101 +24,67 @@ typedef NS_ENUM(NSUInteger, NYTViewControllerPhotoIndex) {
 @interface NYTViewController () <NYTPhotosViewControllerDelegate>
 
 @property (weak, nonatomic) IBOutlet UIButton *imageButton;
-@property (nonatomic) NSArray *photos;
+@property (nonatomic) NYTPhotoViewerArrayDataSource *dataSource;
 
 @end
 
 @implementation NYTViewController
 
 - (IBAction)imageButtonTapped:(id)sender {
-    self.photos = [[self class] newTestPhotos];
-    
-    NYTPhotosViewController *photosViewController = [[NYTPhotosViewController alloc] initWithPhotos:self.photos initialPhoto:nil delegate:self];
+    self.dataSource = [self.class newTimesBuildingDataSource];
+
+    NYTPhotosViewController *photosViewController = [[NYTPhotosViewController alloc] initWithDataSource:self.dataSource initialPhoto:nil delegate:self];
 
     [self presentViewController:photosViewController animated:YES completion:nil];
     
-    [self updateImagesOnPhotosViewController:photosViewController afterDelayWithPhotos:self.photos];
+    [self updateImagesOnPhotosViewController:photosViewController afterDelayWithDataSource:self.dataSource];
+//    [self switchDataSourceOnPhotosViewController:photosViewController afterDelayWithDataSource:self.dataSource];
 }
 
 // This method simulates previously blank photos loading their images after some time.
-- (void)updateImagesOnPhotosViewController:(NYTPhotosViewController *)photosViewController afterDelayWithPhotos:(NSArray *)photos {
+- (void)updateImagesOnPhotosViewController:(NYTPhotosViewController *)photosViewController afterDelayWithDataSource:(NYTPhotoViewerArrayDataSource *)dataSource {
+    if (dataSource != self.dataSource) {
+        return;
+    }
+
     CGFloat updateImageDelay = 5.0;
     dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(updateImageDelay * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
-        for (NYTExamplePhoto *photo in photos) {
+        for (NYTExamplePhoto *photo in dataSource.photos) {
             if (!photo.image && !photo.imageData) {
                 photo.image = [UIImage imageNamed:@"NYTimesBuilding"];
-                [photosViewController updateImageForPhoto:photo];
+                photo.attributedCaptionSummary = [[NSAttributedString alloc] initWithString:@"Photo which previously had a loading spinner (reopen the photo viewer and scroll to here to see it)" attributes:@{NSForegroundColorAttributeName: [UIColor whiteColor], NSFontAttributeName: [UIFont preferredFontForTextStyle:UIFontTextStyleBody]}];
```

Comment:
Perhaps change the caption to... "Photo which previously had a loading spinner (to see the spinner, reopen the photo viewer and scroll to this photo)"